Sources/StreamChatUI/ChatMessageList/ChatMessageListVC.swift

```diff
@@ -325,24 +350,10 @@ open class ChatMessageListVC:
 
         cell.messageContentView?.delegate = self
         cell.messageContentView?.content = message
-        cell.dateSeparatorView.isHidden = true
-
-        guard isDateSeparatorEnabled, let currentMessage = message else {
-            return cell
-        }
-
-        let currentDay = DateFormatter.messageListDateOverlay.string(from: currentMessage.createdAt)
-        cell.dateSeparatorView.content = currentDay
-
-        // Only show the separator if the previous message is from a different day
-        // TODO: simplify this and make it simpler to customize the logic (ie. extract this into a open `showDateSeparatorView` method)
-        let previousIndexPath = IndexPath(row: indexPath.row + 1, section: indexPath.section)
-        if let previousMessage = dataSource?.chatMessageListVC(self, messageAt: previousIndexPath) {
-            let previousDay = DateFormatter.messageListDateOverlay.string(from: previousMessage.createdAt)
-            cell.dateSeparatorView.isHidden = previousDay == currentDay
-        } else {
-            // If previous message doesn't exist show the separator as well
-            cell.dateSeparatorView.isHidden = false
+        
+        if let currentMessage = message {
```

Comment:
Hm, just noticed that we're working with the optional message. Why don't we do an early return if it's `nil`?

Reply:
Yes, I noticed that as well. But didn't want to change it since I don't know if this is actually on purpose. 🤔 But AFAIK, there's no use case to not returning a message right? The UI would not render anything?

Reply:
@evsaev Fixed it ✅